### PR TITLE
Remmina

### PR DESCRIPTION
**Pull request by kapsh**

### CODE REVIEW

**Comment by kapsh** [comment]
Patch was merged by upstream so I have deleted it. Dependencies and cmake options adjusted to current makefile.

File: packages/net-misc/remmina/remmina.exlib [hunk 1]
```diff
@@ -12,21 +12,30 @@ Remmina is a remote desktop client written in GTK+, aiming to be useful for syst
 HOMEPAGE+=" http://remmina.org/"
 
 LICENCES="GPL-2"
-MYOPTIONS=""
+MYOPTIONS="ssh rdp vnc"
```
Comment (hasufell) [inline packages/net-misc/remmina/remmina.exlib]: Why do we introduce these options? You have a usecase to disable any of them?

Comment (kapsh) [reply packages/net-misc/remmina/remmina.exlib]: Sure, let me explain it.

Personally I don't need an SSH client in this application, and especially when it triggers dependency on libvte. I see VTE-based terminals as pretty basic and slow en masse, so why keep (and periodically update, etc.) reduntant single-dependent package for unused feature?
I am using Remmina only as RDP client. But of course I can presume that others don't want this feature with it's dependency installed (as SSH for me).
And VNC done in the same way just for uniformity.

Comment (hasufell) [reply packages/net-misc/remmina/remmina.exlib]: > And VNC done in the same way just for uniformity.

I don't know what that means. So you have a use case for disabling `ssh`, the other options then are not really needed afais. Keep in mind that options always increase complexity (of configuration, maintenance, ...)

Comment (kapsh) [reply packages/net-misc/remmina/remmina.exlib]: > I don't know what that means.

I was trying to say that if allow to choose single protocol then choosing others too seems to be reasonable. (Actually I have no need in VNC too and disabled it after testing build, but I bear so much personal antipathy only to libvte)

However I can understand your point about increasing complexity and will not insist if you don't wanna to have it here.

Comment (hasufell) [reply packages/net-misc/remmina/remmina.exlib]: So let's keep ssh only

Comment (kapsh) [reply packages/net-misc/remmina/remmina.exlib]: Fine, I've ammended adjusted version.

File: packages/net-misc/remmina/remmina.exlib [hunk 2]
```diff
@@ -12,21 +12,26 @@ Remmina is a remote desktop client written in GTK+, aiming to be useful for syst
...
-        net-libs/libssh
-        net-libs/libvncserver
+        dev-libs/libsecret
         x11-libs/gtk+:3
         x11-libs/libxkbfile
+        ssh? (
+            net-libs/libssh
+            dev-libs/vte:2.91
+        )
+        net-libs/libvncserver
```
Comment (hasufell) [inline packages/net-misc/remmina/remmina.exlib]: alphabetical ordering is off

File: packages/net-misc/remmina/remmina.exlib [hunk 3]
```diff
@@ -36,18 +41,19 @@ CMAKE_SRC_CONFIGURE_PARAMS=(
...
-    -DWITH_LIBSSH=ON
     -DWITH_TRANSLATIONS=ON
-    -DWITH_VTE=ON
-    -DWITH_ZLIB=ON
+    -DWITH_FREERDP=ON
+    -DWITH_LIBVNCSERVER=ON
 
     # off
     -DWITH_APPINDICATOR=OFF
+    -DWITH_TELEPATHY=OFF
```
Comment (hasufell) [inline packages/net-misc/remmina/remmina.exlib]: again, you did something to the alphabetical ordering

File: packages/net-misc/remmina/remmina.exlib [hunk 4]
```diff
@@ -36,18 +41,19 @@ CMAKE_SRC_CONFIGURE_PARAMS=(
     -DWITH_GCRYPT=ON
     -DWITH_GETTEXT=ON
     -DWITH_LIBSECRET=ON
-    -DWITH_LIBSSH=ON
     -DWITH_TRANSLATIONS=ON
-    -DWITH_VTE=ON
-    -DWITH_ZLIB=ON
```
Comment (hasufell) [inline packages/net-misc/remmina/remmina.exlib]: why is `WITH_ZLIB` removed

File: packages/net-misc/remmina/remmina.exlib [hunk 5]
```diff
@@ -36,18 +41,19 @@ CMAKE_SRC_CONFIGURE_PARAMS=(
...
+    -DWITH_LIBVNCSERVER=ON
 
     # off
     -DWITH_APPINDICATOR=OFF
+    -DWITH_TELEPATHY=OFF
+    # NB: upstream has marked avahi and spice as recommended
     -DWITH_AVAHI=OFF
-    -DWITH_FREERDP=OFF
     -DWITH_SPICE=OFF
-    -DWITH_SURVEY=OFF
```
Comment (hasufell) [inline packages/net-misc/remmina/remmina.exlib]: why is `WITH_SURVEY` removed?

Comment (kapsh) [reply packages/net-misc/remmina/remmina.exlib]: Upstream removed it. Please check rcgit17 here https://github.com/FreeRDP/Remmina/blob/next/CHANGELOG.md

File: packages/net-misc/remmina/remmina.exlib (hunk 4 above)
Comment (kapsh) [reply packages/net-misc/remmina/remmina.exlib]: I ran cmake with the same parameters as in exlib and it warned me about unused variables: https://paste.pound-python.org/show/p2Jq2iaeXAtX5m6SpHfR/ 
zlib was cleaned from makefile somewhere around here https://github.com/FreeRDP/Remmina/pull/1263/files

File: packages/net-misc/remmina/remmina.exlib (hunk 3 above)
Comment (kapsh) [reply packages/net-misc/remmina/remmina.exlib]: You got me here, I left some mess after editing. It should be fixed now.

**Comment by kapsh** [comment]
Would you kindly to take another look at this PR?

**Comment by hasufell** [comment]
yeah, I'm slacking like shit... lol

File: packages/net-misc/remmina/remmina.exlib [hunk 6]
```diff
@@ -12,42 +12,47 @@ Remmina is a remote desktop client written in GTK+, aiming to be useful for syst
...
     	dev-util/intltool
     	sys-devel/gettext
     	virtual/pkg-config
     build+run:
+        core/json-glib
         dev-libs/glib:2
         dev-libs/libgcrypt
-        dev-libs/vte:2.91
-        net-libs/libssh
+        dev-libs/libsecret
```
Comment (hasufell) [inline packages/net-misc/remmina/remmina.exlib:26]: libsecret has a non-null slot, namely `:1`

File: packages/net-misc/remmina/remmina.exlib [hunk 7]
```diff
@@ -12,42 +12,47 @@ Remmina is a remote desktop client written in GTK+, aiming to be useful for syst
 HOMEPAGE+=" http://remmina.org/"
 
 LICENCES="GPL-2"
-MYOPTIONS=""
+MYOPTIONS="ssh"
 
 DEPENDENCIES="
```
Comment (hasufell) [inline packages/net-misc/remmina/remmina.exlib:17]: it appears `gnome-desktop/libsoup:2.4` is missing as a dep

**Comment by kapsh** [comment]
Thank you. @hasufell what is your way to restore missing dependencies? Reading configure output? ldd?

**Comment by hasufell** [comment]
> what is your way to restore missing dependencies? Reading configure output? ldd?

1. reading/diffing build system
2. a very shitty bash script, I think there was a better one running on gerrit:

```sh
rundepend() {
	local FILELIST=$(cave print-id-contents --no-all --no-best -t f  --format '%n\n' $@)
	local i
	local f
	local p
	if [[ -z $FILELIST ]] ; then
		FILELIST="$@"
	fi

	echo -e 'Check RDEPENDs of package...'
	filedeps=$(echo $(for i in $(scanelf -L -n -q -F '%n #F' ${FILELIST} | tr , ' ') ; do
		echo ${i}
	done | sort -u))

	for p in $(for f in ${filedeps} ; do
		cave print-owners ${f}
	done | sort -u) ; do
		echo $p
	done
}
```

**Comment by kapsh** [comment]
I see. I also found out that CI runs something like this now https://gitlab.exherbo.org/exherbo-misc/exherbo-dev-tools/blob/master/mscan2.rb
But can't understand a single line of it.

**Pull request merged**